mbed_host_tests/host_tests_conn_proxy/conn_primitive_serial.py

```diff
@@ -101,11 +103,11 @@ def write(self, payload, log=False):
                 self.serial.write(payload)
                 if log:
                     self.logger.prn_txd(payload)
+                return payload
         except SerialException as e:
             self.serial = None
             self.LAST_ERROR = "connection lost, serial.write(%d bytes): %s"% (len(payload), str(e))
             self.logger.prn_err(str(e))
-        return payload
```

Comment:
Also lets not return the same payload. As we either return the same payload or ```None```. Returning the payload perceived as written data makes the caller to put checks like ```if ret == data```. Though effectively it is boolean.
Hence return type can be bool. ```True``` if write is successful. ```False``` if write raises exception.